tqdm/std.py

```diff
@@ -126,6 +127,38 @@ def create_th_lock(cls):
 TqdmDefaultWriteLock.create_th_lock()
 
 
+class CallbackIOWrapper(object):
+    def __getattr__(self, name):
+        return getattr(self._wrapped, name)
+
+    def __setattr__(self, name, value):
+        return setattr(self._wrapped, name, value)
+
+    def __init__(self, callback, stream, method="read"):
+        """
+        Wrap a given `file`-like object's `read()` or `write()` to report
+        lengths to the given `callback`
```

Comment:
Should probably at least document that it doesn't work for generators or anything else that doesn't work with `len()`?